core/src/main/scala/kafka/admin/AclCommand.scala

```diff
@@ -153,6 +256,32 @@ object AclCommand extends Logging {
     authorizer.getAcls()
       .filter { case (resource, acl) => filter.matches(resource.toPattern) }
 
+  private def listAclWithAdminAPI(opts: AclCommandOptions): Unit = {
+    withAdminClient(opts) { adminClient =>
+      val filters = getResourceFilter(opts, dieIfNoResourceFound = false)
+      val resourceToAcls = getAcls(adminClient, filters)
+
+      for ((resource, acls) <- resourceToAcls)
+        println(s"Current ACLs for resource `$resource`: $Newline ${acls.map("\t" + _).mkString(Newline)} $Newline")
+    }
+  }
+
+  private def getAcls(adminClient: JAdminClient, filters: Set[ResourcePatternFilter]): Map[ResourcePattern, Set[AccessControlEntry]] = {
+    val aclBindings =
+      if (filters.isEmpty) adminClient.describeAcls(AclBindingFilter.ANY).values().get().asScala.toList
+      else {
+        val results = for (filter <- filters) yield {
```

Comment:
Since you have multiple inputs, please use the batch API here rather than sending lots of single-entry RPCs

Reply:
looks like we don't have batchAPI for describing acls for multiple ResourcePatternFilters

```
describeAcls(AclBindingFilter filter, DescribeAclsOptions options);
```